xonsh/xontribs.py

```diff
@@ -36,7 +36,10 @@ def xontrib_context(name):
                           ImportWarning)
         return {}
     m = importlib.import_module(spec.name)
-    ctx = {k: getattr(m, k) for k in dir(m) if not k.startswith('_')}
+    if hasattr(m, '__all__'):
```

Comment:
Not that this is a huge deal, and not blocking, but I think you could save a lookup in m by doing:

```python
allnames = getattr(m, '__all__', None)
if allnames is None:
    ctx = {k: getattr(m, k) for k in dir(m) if not k.startswith('_')}
else:
    ctx = {k: getattr(m, k) for k in allnames}
```